src/ini.zig

```diff
@@ -0,0 +1,1076 @@
+const std = @import("std");
+const bun = @import("root").bun;
+const Allocator = std.mem.Allocator;
+const E = bun.JSAst.E;
+const Expr = bun.JSAst.Expr;
+const Loc = bun.logger.Loc;
+const js_ast = bun.JSAst;
+const Rope = js_ast.E.Object.Rope;
+const Output = bun.Output;
+const Global = bun.Global;
+
+pub const Parser = struct {
+    opts: Options = .{},
+    source: bun.logger.Source,
+    src: []const u8,
+    out: Expr,
+    logger: bun.logger.Log,
+    arena: std.heap.ArenaAllocator,
+    env: *bun.DotEnv.Loader,
+
+    const Options = struct {
+        bracked_array: bool = true,
+    };
+
+    pub fn init(allocator: Allocator, path: []const u8, src: []const u8, env: *bun.DotEnv.Loader) Parser {
+        return .{
+            .logger = bun.logger.Log.init(allocator),
+            .src = src,
+            .out = Expr.init(E.Object, E.Object{}, Loc.Empty),
+            .source = bun.logger.Source.initPathString(path, src),
+            .arena = std.heap.ArenaAllocator.init(allocator),
+            .env = env,
+        };
+    }
+
+    pub fn deinit(this: *Parser) void {
+        this.logger.deinit();
+        this.arena.deinit();
+    }
+
+    pub fn parse(this: *Parser) !void {
+        try this.parseImpl();
+    }
+
+    inline fn shouldSkipLine(line: []const u8) bool {
+        if (line.len == 0 or
+            // comments
+            line[0] == ';' or
+            line[0] == '#') return true;
+
+        // check the rest is whitespace
+        for (line) |c| {
+            switch (c) {
+                ' ', '\t', '\n', '\r' => {},
+                '#', ';' => return true,
+                else => return false,
+            }
+        }
+        return true;
+    }
+
+    fn parseImpl(this: *Parser) !void {
+        var iter = std.mem.splitScalar(u8, this.src, '\n');
+        var head: *E.Object = this.out.data.e_object;
+
+        // var duplicates = std.StringArrayHashMapUnmanaged(u32){};
+        // defer duplicates.deinit(allocator);
+
+        var stack = std.heap.stackFallback(@sizeOf(Rope) * 6, this.arena.allocator());
```

Comment:
Can we make it more clear this is exclusively for use by section names?
```suggestion
        var rope_only_stack_allocator = std.heap.stackFallback(@sizeOf(Rope) * 6, this.arena.allocator());
```